builder/vmware/iso/builder.go

```diff
@@ -215,12 +215,8 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 	state.Put("hook", hook)
 	state.Put("ui", ui)
 
-	steps := []multistep.Step{
-		&vmwcommon.StepPrepareTools{
-			RemoteType:        b.config.RemoteType,
-			ToolsUploadFlavor: b.config.ToolsUploadFlavor,
-		},
-		&common.StepDownload{
+	stepVerifyCache := &stepVerifyCache{
```

Comment:
I think I understand what you're doing, here, but I don't think this is the way to do it. I think you can just modify `stepRemoteUpload` to do something like

```
remotePath := esx5.cachePath(state.Get(s.ResultKey))
if esx5.verifyChecksum(s.download.ChecksumType, s.download.Checksum, remotePath) {
    return multistep.ActionContinue
}
```